rbac/migration_tool/migrate.py

```diff
@@ -123,26 +139,46 @@ def migrate_data_for_tenant(tenant: Tenant, exclude_apps: list, write_relationsh
         # This must always be the case, but this should at least start failing you if you forget.
         BindingMapping.objects.bulk_create(mappings, ignore_conflicts=False)
 
-        output_relationships(tuples, write_relationships)
+        replicator.replicate(
+            ReplicationEvent(
+                event_type=ReplicationEventType.MIGRATE_CUSTOM_ROLE,
+                info={"role_uuid": str(role.uuid)},
+                partition_key="rbactodo",
+                add=tuples,
+            )
+        )
 
         logger.info(f"Migration completed for role: {role.name} with UUID {role.uuid}.")
     logger.info(f"Migrated {roles.count()} roles for tenant: {tenant.org_id}")
 
 
-def migrate_data(exclude_apps: list = [], orgs: list = [], write_relationships: bool = False):
+def migrate_data(exclude_apps: list = [], orgs: list = [], write_relationships: str = "False"):
     """Migrate all data for all tenants."""
     count = 0
     tenants = Tenant.objects.exclude(tenant_name="public")
+    replicator = _get_replicator(write_relationships)
     if orgs:
         tenants = tenants.filter(org_id__in=orgs)
     total = tenants.count()
     for tenant in tenants.iterator():
         logger.info(f"Migrating data for tenant: {tenant.org_id}")
         try:
-            migrate_data_for_tenant(tenant, exclude_apps, write_relationships)
+            migrate_data_for_tenant(tenant, exclude_apps, replicator)
         except Exception as e:
             logger.error(f"Failed to migrate data for tenant: {tenant.org_id}. Error: {e}")
             raise e
         count += 1
         logger.info(f"Finished migrating data for tenant: {tenant.org_id}. {count} of {total} tenants completed")
     logger.info("Finished migrating data for all tenants")
+
+
+def _get_replicator(write_relationships: str) -> RelationReplicator:
```

Comment:
This is mixing a boolean-like input with  "sync" and "async" options. 

It would require changing the `write_relationships` parameter in caller methods to something more options-oriented or convert it to proper option before ` _get_replicator` is called.

It is not big thing, but it could improve consistency.

Reply:
Keeping True/False was for compatibility, particularly with the QA test suite. All the caller methods should have been changed I think. Did I miss any?

Reply:
This is OK I saw connections with  QA test suite.